tests/templates/test_subroutines/test_qsvt.py

```diff
@@ -0,0 +1,173 @@
+
+    @pytest.mark.parametrize(
+        ("U_A", "lst_projectors", "wires", "results"),
+        [
+            (
+                qml.BlockEncode(0.1, wires=0),
+                [qml.PCPhase(0.2, dim=1, wires=0), qml.PCPhase(0.3, dim=1, wires=0)],
+                0,
+                [
+                    qml.BlockEncode(np.array([[0.1]]), wires=[0]),
+                    qml.PCPhase(0.2, dim=2, wires=[0]),
+                    qml.adjoint(qml.BlockEncode(np.array([[0.1]]), wires=[0])),
+                    qml.PCPhase(0.3, dim=2, wires=[0]),
+                ],
+            ),
+            (
+                qml.PauliZ(wires=0),
+                [qml.RZ(0.1, wires=0), qml.RY(0.2, wires=0), qml.RZ(0.3, wires=1)],
+                [0, 1],
+                [
+                    qml.PauliZ(wires=[0]),
+                    qml.RZ(0.1, wires=[0]),
+                    qml.adjoint(qml.PauliZ(wires=[0])),
+                    qml.RY(0.2, wires=[0]),
+                    qml.PauliZ(wires=[0]),
+                    qml.RZ(0.3, wires=[1]),
+                ],
+            ),
+        ],
+    )
+    def test_queuing_ops(self, U_A, lst_projectors, wires, results):
+        with qml.tape.QuantumTape() as tape:
+            qml.QSVT(U_A, lst_projectors, wires)
+
+        for idx, val in enumerate(tape.expand().operations):
+            assert val.name == results[idx].name
+            assert val.parameters == results[idx].parameters
+
+    def test_queuing_ops_defined_in_circuit(self):
+        lst_projectors = [qml.PCPhase(0.2, dim=1, wires=0), qml.PCPhase(0.3, dim=1, wires=0)]
+        wires = [0, 1]
+        results = [
+            qml.PauliX(wires=[0]),
+            qml.PCPhase(0.2, dim=1, wires=[0]),
+            qml.adjoint(qml.PauliX(wires=[0])),
+            qml.PCPhase(0.3, dim=1, wires=[0]),
+        ]
+
+        with qml.tape.QuantumTape() as tape:
+            qml.QSVT(qml.PauliX(wires=0), lst_projectors, wires)
+
+        for idx, val in enumerate(tape.expand().operations):
+            assert val.name == results[idx].name
+            assert val.parameters == results[idx].parameters
+
+    def test_queuing_callables(self):
+        def my_qfunc(A):
+            return qml.PauliX(wires=0)
+
+        def lst_phis(phis):
+            return [qml.PCPhase(i, 2, wires=[0, 1]) for i in phis]
+
+        A = np.array([[0.1, 0.2], [0.3, 0.4]])
+        phis = np.array([0.2, 0.3])
+
+        with qml.tape.QuantumTape() as tape:
+            qml.QSVT(my_qfunc(A), lst_phis(phis), wires=[0, 1])
+
+        results = [
+            qml.PauliX(wires=[0]),
+            qml.PCPhase(0.2, dim=2, wires=[0]),
+            qml.adjoint(qml.PauliX(wires=[0])),
+            qml.PCPhase(0.3, dim=2, wires=[0]),
+        ]
+
+        for idx, val in enumerate(tape.expand().operations):
+            assert val.name == results[idx].name
+            assert val.parameters == results[idx].parameters
+
+    def test_queuing_callables2(self):
```

Comment:
Could this be merged with the previous test using `@pytest.mark.parametrize`?

Reply:
Merged.